### PR TITLE
Added support for paths with spaces, and bats tests.

### PR DESCRIPTION
**Pull request by beverage**
The maven wrapper script currently does not work in folder heirarchies that contain spaces.  The simple fix was to quote the $(pwd) statements in find_maven_basedir along with some more quote-fu on the MAVEN_PROJECTBASEDIR and MAVEN_OPTS line, but I opted for a change that would be testable instead.

This involved parameterizing find_maven_basedir, and reorganizing the script to make the actual body of it excludable by bats when it is loaded.

My apologies as this will make the PR a bit messier than need be for this change, but I will do my best to highlight the most relevant bits.
- Line 39 - relocated find_maven_basedir followed by concat_lines
- Line 66 - test to prevent script execution when loaded by bats
- Lines 218 and 223 - using the updated find method

I did incorporate the bats run as part of the automation launched when building yet as that would require a dependency on an external toolset and should be probably be discussed first.

Tested with bats tests, and manually on the local projects here that first exposed the issue.

### CODE REVIEW

**Comment by beverage** [comment]
Also, incorporated the changes in https://github.com/takari/maven-wrapper/pull/21 as I was already dealing with pathing.

**Comment by beverage** [comment]
Possibly related to https://github.com/takari/maven-wrapper/issues/23

File: mvnw.cmd [hunk 1]
```diff
@@ -118,7 +118,7 @@ for /F "usebackq delims=" %%a in ("%MAVEN_PROJECTBASEDIR%\.mvn\jvm.config") do s
 
 SET MAVEN_JAVA_EXE="%JAVA_HOME%\bin\java.exe"
 
-set WRAPPER_JAR=""%MAVEN_PROJECTBASEDIR%\.mvn\wrapper\maven-wrapper.jar""
+set WRAPPER_JAR="%MAVEN_PROJECTBASEDIR%\.mvn\wrapper\maven-wrapper.jar"
```
Comment (chipkillmar) [inline mvnw.cmd:121]: Without this change, I'm seeing the following error on Windows when running `mvnw.cmd` in a directory name that contains spaces:

```
Error: Could not find or load main class With Spaces\.mvn\wrapper\maven-wrapper.jar
```

:+1:

**Comment by mosabua** [comment]
Replaced with simple commit of fix in https://github.com/takari/maven-wrapper/commit/2fea7af95512940335c593751517abf67beab1f5

**Pull request closed without merging**